Fix double-transition bug

domq opened the pull request:
Fixes #26451

By applying the transition: CSS property only to classes that are
active during sliding, we avoid an unnecessary, non-zero-time
animation that although mostly invisible, does interfere with e.g.
z-index based parallax carousels (see
https://codepen.io/domq/pen/ZoJrQZ)

domq committed:
Fix double-transition bug

Fixes #26451

By applying the transition: CSS property only to classes that are
active during sliding, we avoid an unnecessary, non-zero-time
animation that although mostly invisible, does interfere with e.g.
z-index based parallax carousels (see
https://codepen.io/domq/pen/ZoJrQZ)

XhmikosR reviewed (approved):
LGTM

andresgalante reviewed (changes requested):
Hi @domq thanks for the contribution.
I built your branch and the slide animation is not working.
Am I missing something?

domq commented:
@andreasgalante I'm sorry that the slide animation is not working for you; could you please let me know the steps you took for trying it out? Here is what I did try:

- `npm i`
- `npm test` (I did get failures, but they are unrelated to my change afaict)
- opened `js/tests/visual/carousel.html` in browser — everything seems OK

domq commented:
![sliding](https://user-images.githubusercontent.com/1629585/40011494-4657c0ee-57a8-11e8-9ac9-af9d8a2e424c.gif)

andresgalante commented:
Sorry, you are right it is working when I open `js/tests/visual/carousel.html` directly.

But when I run the docs `npm run docs-serve` it's not working for me on  `http://localhost:9001/docs/4.1/components/carousel/`

It might be just me, but just wanted to make sure.

XhmikosR commented:
@andresgalante: unless you run `npm run dist` or `js` the dist files used by the docs are the old ones.

domq commented:
Thanks @andresgalante for the review. Just to be sure, is the defect you are witnessing the same as in the gif below? If so, I fixed it and rebased onto today's `v4-dev`, kindly take another look.

![actual](https://user-images.githubusercontent.com/1629585/40222117-c77b0c20-5a7e-11e8-902d-a1bb951fac36.gif)

I.e. no smooth transition at http://localhost:9001/docs/4.1/components/carousel/, as opposed to the online docs (https://getbootstrap.com/docs/4.1/components/carousel/) which work as expected:

![expected](https://user-images.githubusercontent.com/1629585/40222102-bdca2418-5a7e-11e8-897c-aad0f8995ec1.gif)

andresgalante commented:
I am sorry @domq it was my build 😞

andresgalante commented:
oh! I just read your last comment, thanks for fixing it.

I though it was my build but your last commit is doing the trick :)
